examples/common/python/connectors/direct/tcs_listener/tcs_worker_registry_handler.py

```diff
@@ -163,54 +144,49 @@ def __validate_input_worker_status(self, input_json):
             return False
 
 # ------------------------------------------------------------------------------------------------
-    def __process_worker_set_status(self, worker_id, input_value, response):
+    def WorkerSetStatus(self, **params):
         """
         Function to set the status of worker
         Parameters:
             - worker_id is the worker id specified in the input json str.
-            - input_json_str is a worker request json as per TCF API 5.3.3 Worker Set Status JSON Payload
+            - input_json_str is a worker request json as per TCF API 5.3.3
+                Worker Set Status JSON Payload
             - response is the response object to be returned to client.
         """
 
         # status can be one of active, offline, decommissioned, or compromised
 
-        jrpc_id = input_value["id"]
+        worker_id = must_get_worker_id(params)
 
         value = self.kv_helper.get("workers", worker_id)
         if value:
             json_dict = json.loads(value)
-            if self.__validate_input_worker_status(input_value) == False:
-                response = utility.create_error_response(
-                           WorkerError.INVALID_PARAMETER_FORMAT_OR_VALUE,
-                           jrpc_id,
-                           "Invalid parameter: worker status code")
-                return response
-
-            json_dict['status'] = input_value['params']['status']
+            if not self.__validate_input_worker_status(params):
+                raise JSONRPCDispatchException(
+                    WorkerError.INVALID_PARAMETER_FORMAT_OR_VALUE,
+                    "Invalid parameter: worker status code")
+
+            json_dict['status'] = params['status']
             value = json.dumps(json_dict)
             self.kv_helper.set("workers", worker_id, value)
-            response = utility.create_error_response(
```

Comment:
It should return functionality not implemented.

Reply:
Do you mean replacing the `SUCCESS` with something like `FUNCTIONALITY_NOT_IMPLEMENTED`?